src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -1720,9 +1732,10 @@ public enum FunctionInvocationStatus
         Exception,
     }
 
-    private struct ApprovalResultWithRequestMessage
+    private readonly struct ApprovalResultWithRequestMessage
     {
-        public FunctionApprovalResponseContent Response { get; set; }
-        public ChatMessage? RequestMessage { get; set; }
+        public FunctionApprovalResponseContent Response { get; init; }
+        public FunctionCallContent FunctionCallContent { get; init; }
```

Comment:
Why is this needed? Is that just to make accessing this less verbose? Otherwise isn't this just Response.CallContent?

Reply:
Because Response.CallContent is now AIContent, I would have to do a cast or type check later if I don't save it here.

Reply:
> Is that just to make accessing this less verbose?

Yes but not for the sake of using less dots/properties.

Reply:
It wasn't for perf, it felt less risky to have the concrete-type property and leverage that the initialization already validated it was a `FunctionCallContent`.  I applied your suggestion anyway since is functionally the same.